integration-cli/docker_cli_build_test.go

```diff
@@ -5920,6 +5951,51 @@ func (s *DockerSuite) TestBuildCopyFromPreviousFrom(c *check.C) {
 	c.Assert(strings.TrimSpace(out), check.Equals, "def")
 }
 
+func (s *DockerSuite) TestBuildCopyFromImplicitFrom(c *check.C) {
+	dockerfile := `
+		FROM busybox
+		COPY --from=busybox /etc/passwd /mypasswd
+		RUN cmp /etc/passwd /mypasswd`
```

Comment:
@tonistiigi I made certain restrictions for Windows in https://github.com/docker/docker/pull/32084 that we did not thought of previously. Basically, I made it mandatory for source paths to be both absolute and drive qualified.
Thus `COPY --from=0 /test test` won't work but `COPY --from=0 c:/test test` will.
I think I can soften the rule a little (authorize relative paths if they contain at least one path segment and this segment is not `windows`). That would make it easier to write x-plat tests.